packages/rollup-plugin/src/hbUtils.js

```diff
@@ -0,0 +1,132 @@
+import originalFetch, { FormData, fileFrom } from 'node-fetch'
+import fetchRetry from 'fetch-retry';
+const fetch = fetchRetry(originalFetch)
+
+/******************************
+ * Everything in this file is designed to be shared with the webpack plugin
+ * e.g. by removing specifics about how the bundle is formatted 
+ * In a follow-up, we can extract this into a module to share among the plugins
+*******************************/
+
+/**
+ * Uploads sourcemaps to API endpoint
+ *
+ * @param {Array} sourcemapData An array of sourcemap data, each entry
+ *   should look like { sourcemapFilename, sourcemapFilePath, jsFilename, jsFilePath }
+ * @param {Object} hbOptions See ./options.js 
+ * @returns {Promise} Resolves if all sourcemaps are uploaded
+ */
+export async function uploadSourcemaps({ sourcemapData = [], hbOptions }) {
+  if (sourcemapData.length === 0 && !hbOptions.silent) {
+    console.warn('Could not find any sourcemaps in the bundle. Nothing will be uploaded.')
+  }
+
+  const sourcemapUploadPromises = sourcemapData.map(data => {
+    return uploadSourcemap({ 
+      ...hbOptions,
+      ...data
+    })
+  })
+  
+  return Promise.all(sourcemapUploadPromises)
```

Comment:
I'm wondering if we can use [Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) here.

Reply:
Good point! I wasn't sure if we want the failure of an upload to cause the user's build to fail, or if we want to just log to the console any failures. Current behavior will cause the build to fail if any of the uploads fail. `allSettled` would allow the build to pass, and then we could just log out what happened. I could see an argument for either behavior... having the builds fail is actually pretty helpful when you're first setting it up because it's super obvious that something's broken. But if your CI started failing just because of sourcemap upload, it would probably be pretty annoying.

Reply:
Interesting points 🤔 . I think we can let the build fail if the source map file upload fails.
To me, this is an important part of the deployment; otherwise reported errors with the new release will not show up as expected, and it might take time until people realise that the upload is failing.

However, my original suggestion for `Promise.allSetlled` is whether we want to show more detailed error messages. For example, `Promise.all` will reject with the first promise that failed, but we won't know the state of the other promises that haven't resolved/rejected yet. It may be unnecessary information, though. Like, is there any additional value if we tell "these 10 source map file uploads failed" or just that "source map files upload failed" ?

Reply:
Just added logging for the number of successes, and re-threw the failures in order to cause a failure of the build!